app/controllers/v1/pacbio/libraries_controller.rb

```diff
@@ -38,11 +38,25 @@ def library
 
       def params_names
         params.require(:data).require(:attributes)[:libraries].map do |param|
-          param.permit(:volume, :concentration, :library_kit_barcode, :fragment_size,
-                       :pacbio_tag_id, :pacbio_request_id).to_h
+          library_params_names(param)
+        end.flatten
+      end
+
+      def library_params_names(params)
+        params.permit(:volume, :concentration, :library_kit_barcode, :fragment_size, :relationships)
+              .to_h.tap do |library|
+          library[:requests] = request_param_names(params)
         end
       end
 
+      def request_param_names(params)
+        params.require(:relationships)[:requests].require(:data).map do |request|
+          request.permit(:id, :type).to_h.tap do |tag|
+            tag[:tag] = request.require(:relationships)[:tag].require(:data).permit(:id, :type).to_h
```

Comment:
Maybe we could pull `request.require(:relationships)[:tag].require(:data).permit(:id, :type).to_h`  out to `tag_params_names` method for consistency and readability

Reply:
agreed. Done. It might be a good idea to abstract the behaviour out for params names as there is quite a lot of it in the controllers. I will add it as an issue.